### PR TITLE
[PINOT-3505] Update committing segmentmetadata from the incoming metadata file

### PR DESCRIPTION
**Pull request by mcvsubbu**
Also fixed record count in RealtimeSegmentDataManager

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java [hunk 1]
```diff
@@ -358,6 +369,32 @@ public boolean commitSegment(String rawTableName, final String committingSegment
     oldSegMetadata.setEndTime(now);
     oldSegMetadata.setDownloadUrl(
         ControllerConf.constructDownloadUrl(rawTableName, committingSegmentNameStr, _controllerConf.generateVipUrl()));
+    // Pull segment metadata from incoming segment and set it in zk segment metadata
+    final String baseDir = StringUtil.join("/", _controllerConf.getDataDir(), rawTableName);
```
Comment (jfim) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Would be good to add a comment that explains what the path looks like.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java [hunk 2]
```diff
@@ -358,6 +369,32 @@ public boolean commitSegment(String rawTableName, final String committingSegment
...
+      }
+    }
+    oldSegMetadata.setCrc(Long.valueOf(segmentMetadata.getCrc()));
+    oldSegMetadata.setStartTime(segmentMetadata.getTimeInterval().getStartMillis());
+    oldSegMetadata.setEndTime(segmentMetadata.getTimeInterval().getEndMillis());
+    oldSegMetadata.setIndexVersion(segmentMetadata.getVersion());
+    oldSegMetadata.setTotalRawDocs(segmentMetadata.getTotalRawDocs());
+
     final ZNRecord oldZnRecord = oldSegMetadata.toZNRecord();
     final String oldZnodePath = PinotRealtimeSegmentManager.getSegmentsPath() + "/" + realtimeTableName + "/" + committingSegmentNameStr;
```
Comment (jfim) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Would be good to use utility methods for those, I'm pretty sure there's a method that builds a property store znode path given a table and segment name.

**Comment by jfim** [comment]
lgtm

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java (hunk 1 above)
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Done

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java (hunk 2 above)
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Changed to use ZKMetadataProvider all through the file, and removed the getSegmentsPath() method added specifically to get the segments path. We need to do the same in PinotRealtimeSegmentManager as well, at some point.

**Pull request merged**